### PR TITLE
fix(Dropdown): added onClick override on DropdownItem 

### PR DESCRIPTION
**Pull request by silviuaavram**
It allows supplying an override to onClick for a `DropdownItem`.

For now I am not sure if we want a default stopPropagation from our code, open to debate.

### CODE REVIEW

File: packages/react/src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -590,7 +590,8 @@ class Dropdown extends AutoControlledComponent<Extendable<DropdownProps>, Dropdo
               key: (item as any).header,
             }),
         },
-        overrideProps: () => this.handleItemOverrides(item, index, getItemProps),
+        overrideProps: (predefinedProps: DropdownItemProps) =>
+          this.handleItemOverrides(predefinedProps, item, index, getItemProps),
```
Comment (layershifter) [inline packages/react/src/components/Dropdown/Dropdown.tsx]: Please consider to refactor to this:
```jsx
overrideProps: this.handleItemOverrides(item, index, getItemProps),
```
```tsx
  private handleItemOverrides = (
    item: ShorthandValue,
    index: number,
    getItemProps: (options: GetItemPropsOptions<ShorthandValue>) => any),
  => (predefinedProps: DropdownItemProps) => any,
```

**Comment by layershifter** [comment]
Can you please add a UT for this?

File: packages/react/src/components/Dropdown/Dropdown.tsx (hunk 1 above)
Comment (silviuaavram) [reply packages/react/src/components/Dropdown/Dropdown.tsx]: good idea! will update all callbacks similar to this one. and will add unit tests for them, thanks!

File: packages/react/test/specs/components/Dropdown/Dropdown-test.tsx [hunk 2]
```diff
@@ -6,6 +6,7 @@ import Dropdown from 'src/components/Dropdown/Dropdown'
 import DropdownSearchInput from 'src/components/Dropdown/DropdownSearchInput'
 import { isConformant } from 'test/specs/commonTests'
 import { mountWithProvider } from 'test/utils'
+import { DropdownSelectedItem } from '@stardust-ui/react/src'
```
Comment (layershifter) [inline packages/react/test/specs/components/Dropdown/Dropdown-test.tsx]:
```suggestion
import { DropdownSelectedItem } from '@stardust-ui/react'
```
or
```suggestion
import DropdownSelectedItem from 'src/components/Dropdown/DropdownSelectedItem'
```

File: packages/react/test/specs/components/Dropdown/Dropdown-test.tsx [hunk 3]
```diff
@@ -957,4 +958,38 @@ describe('Dropdown', () => {
...
+      const mockedEvent = { stopPropagation }
+      const items = [{ header: 'Venom', onClick }]
+      const wrapper = mountWithProvider(<Dropdown items={items} />)
+      const triggerButton = wrapper.find(`button.${Dropdown.slotClassNames.triggerButton}`)
+
+      triggerButton.simulate('click')
+      const firstItem = wrapper.find(`li.${Dropdown.slotClassNames.item}`)
+      firstItem.simulate('click', mockedEvent)
+
+      expect(onClick).toBeCalledTimes(1)
```
Comment (layershifter) [inline packages/react/test/specs/components/Dropdown/Dropdown-test.tsx:1028]: Please add an assertion for props, i.e. `toBeCalledWith()`

**Review by layershifter — APPROVED** [review]

**Comment by layershifter** [comment]
And please mention in PR description and changelog that we stopped propagate events.

File: packages/react/test/specs/components/Dropdown/Dropdown-test.tsx (hunk 2 above)
Comment (silviuaavram) [reply packages/react/test/specs/components/Dropdown/Dropdown-test.tsx]: this NEVER works by default

**Review by jurokapsiar — APPROVED** [review]

**Pull request merged**